keras_cv/ops/box_matcher.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Argmax-based box matching"
+
+from typing import List
+from typing import Tuple
+
+import tensorflow as tf
+
+
+class ArgmaxBoxMatcher:
+    """Box matching logic based on argmax of highest value (iou).
+
+    This class computes matches from a similarity matrix. Each row will be
+    matched to at least one column, the matched result can either be positive
+     / negative, or simply ignored depending on the setting.
+
+    The settings include `thresholds` and `indicators`, for example if:
+    1) thresholds=[negative_threshold, positive_threshold], and
+       indicators=[negative_value, ignore_value, positive_value]: the rows will
+       be assigned to positive_value if its argmax result is above
+       positive_threshold; the rows will be assigned to negative_value if its
+       argmax result is below negative_threshold, and the rows will be assigned
+       to ignore_value if its argmax result is between negative_threshold and
+       positive_threshold.
+    2) thresholds=[negative_threshold, positive_threshold], and
+       indicators=[ignore_value, negative_value, positive_value]: the rows will
```

Comment:
This class is actually more generic than that, it supports list of any length. Instead of splitting the arguments, I add another example where there's only [positive_threshold] and [negative_value, positive_value] -- so there's no `ignore_value` here.
In the end, we aim to providing default values to the constructor when using it within RetinaNetLabelEncoder and RPNLabelEncoder, acting as examples of how it can be used, and users can customize their own matching logic on top of these two.

Reply:
> This class is actually more generic than that, it supports list of any length. Instead of splitting the arguments, I add another example where there's only [positive_threshold] and [negative_value, positive_value] -- so there's no ignore_value here.


ok, I understand now.  Thanks for explaining this - I misread the api.